lib/core/src/sync/model/data.rs

```diff
@@ -114,6 +136,20 @@ impl SendSyncData {
             }
         }
     }
+
+    pub(crate) fn updated_fields(swap: Option<SendSwap>, update: &SendSwap) -> Option<Vec<String>> {
+        match swap {
+            Some(swap) => {
+                let mut updated_fields = vec![];
+                if !matches!(update.preimage.clone(), Some(u) if swap.preimage.map_or(false, |s| s == u))
```

Comment:
Can we simplify this by just comparing` update.preimage `to `swap.preimage `?

Reply:
I guess the code is trying to update only when there it's `Some`, no need to do so for the none case. So simply doing swap.preimage != update.preimage won't work.
Does 2408bde28deb make sense?